internal/testrunner/runners/system/runner.go

```diff
@@ -747,6 +759,43 @@ func validateFields(docs []common.MapStr, fieldsValidator *fields.Validator, dat
 	return nil
 }
 
+func getDocEventCount(events map[string]interface{}) int {
+	for k, v := range events {
+		if reflect.ValueOf(v).Kind() == reflect.Slice {
+			e := events[k].([]interface{})
```

Comment:
I don't think this function will be needed based on other comments. I still wanted to mention that reflection does not look required. A [type assertion](https://go.dev/ref/spec#Type_assertions) will work.

`if e, ok := v.([]interface{}); ok {`